libfreerdp/core/settings.c

```diff
@@ -701,6 +701,17 @@ static BOOL freerdp_settings_int_buffer_copy(rdpSettings* _settings, const rdpSe
 		_settings->ClientRandomLength = settings->ClientRandomLength;
 	}
 
+	if (settings->ServerCertificateLength)
+	{
+		_settings->ServerCertificate = (BYTE*)malloc(settings->ServerCertificateLength);
+
+		if (!_settings->ServerCertificate)
+			goto out_fail;
+
+		CopyMemory(_settings->ServerCertificate, settings->ServerCertificate, _settings->ServerCertificateLength);
```

Comment:
This looks like a bug, surely you meant `settings->ServerCertificateLength` here? because you copy it only in the next line.